### ISSUE TITLE
Expansion Keyboard API (particularly for mobile)

### ISSUE THREAD

**Issue by nxtexe** [mdn/kuma#7592]
**Summary**
_What problem would this solve?_
The problem this would solve is that on mobile there is no way to programmatically request the use of the soft keyboard which makes it difficult for rendering solutions on the web that do not use html to render UI to get user input.

**Audience**
_Who has this problem? Everyone? Article authors? Etc._
The audience this problem affects is developers looking to leverage wasm to offer near native web experiences. The web is moving to evolve using wasm for performance in combination with javascript. As it stands right now, UI can be drawn using wasm on a html5 canvas, but when it comes to user input on mobile, some weird workarounds have to be used in order to achieve the end goal of getting user input from the soft keyboard.

**Rationale**
_How do you know that people identified above actually have this problem?_
I know the people identified have this problem as I am currently developing an application to render using the html5 canvas, but keyboard input is tricky on mobile, near impossible since I can't get access to the soft keyboard programmatically.

**Workaround**
_How are the people identified above currently handling this problem?_
I currently use a workaround in which I use hidden html elements to solve the problem. I have a top layer element preferably an a tag, that captures touch events and behind the canvas is an input tag. I register a touch event listener on the a tag and inside the event listener callback function I grab the input tag behind the canvas and call it's .click method which brings the keyboard up. This workaround isn't guaranteed to work, especially on safari (iOS) where it doesn't work at all. I've only gotten it to work on android devices using chrome.

**Proposal**
_Do you have suggestions for solving this problem?_
Implementing an expansion of the Keyboard API using Web IDL bindings could solve the issue. I am however aware of the web security issues that this feature poses, but with wasm currently offering the potential to give near native performance, we should find ways to safely provide such a core functionality.

**Additional context**
_Is there anything else we should know?_

**Comment by peterbe** [comment]
Was there a particular page on [MDN Web Docs](https://developer.mozilla.org) where you think this is relevant and where the documentation needs to be improved?

**Comment by nxtexe** [comment]
> Was there a particular page on [MDN Web Docs](https://developer.mozilla.org) where you think this is relevant and where the documentation needs to be improved?

Yes. This page [here](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard).
